doc/xversionmessage.md

```diff
@@ -0,0 +1,250 @@
+XVersionMessage: BCH node extended version and configuration fields
+===================================================================
+
+DRAFT specification
+Version: 0.0.3
+
+Overview
+--------
+
+Using the `version` message in the Bitcoin protocol (for more details,
+see https://en.bitcoin.it/wiki/Protocol_documentation#version), peers
+announce their capabilities and preferences to each other. The
+`version` message is limited in scope to a fixed set of well-known
+fields.
+
+It is deemed desirable to be able to announce further configuration
+and version information to peers. For example, this can then be used
+to version advanced block transmission protocols like Graphene.
+
+Notably, the `version` message can not be extended as there are
+clients on the current BCH network that will consider an oversized
+`version` message illegal (e.g. Bitcoin XT).
+
+For this reason, an additional message type named `xversion` is
+specified here in, plus an acknowledgement named `xverack`. The
+`xversion` message transports a generic key-value map that is meant to
+hold the configuration and version parameters.
+
+As a general reminder, it should be noted that the values given in
+this extended `xversion` version and configuration map are, except for
+basic type checks and mappings, not in any way validated and a node
+can very easily lie about them. Care should be taken that any
+implementation depending for its configuration on the values in the
+xmap will properly account for this fact and ban or otherwise penalize
+nodes that do not conform with their advertised behavior.
+
+## Encoding of the `xversion` and `xverack` message types
+
+The `xversion` message contains a single compound field which is a
+serialized key-value map henceforth named `xmap` that maps 64-bit
+integer values to variable-sized byte vectors. The message itself is
+encoded in the standard Bitcoin message frame that is documented elsewhere.
+
+The `map` is encoded using Bitcoin's usual network de-/serialization
+schemes while using `COMPACT_SIZE` size encodings whenever possible.
+Note that this is different from using the default encoding you would
+get when serializing through the `std::map` serializer (which is
+commonly found in `serialize.h` in most current C++ implementations based
+on the original Satoshi code).
```

Comment:
Fair question. If the table is full (at its limit of 100kB) of (expected) small int values, this would save a couple hundred kB in network traffic for each node connect. I guess this is at the edge of premature optimization. If you have a strong opinion to do it the old way, I can change it.
In any case, I think the complexity of the two approaches are comparable, however.

Reply:
Do you expect the table to become this large?  What's the order of magnitude expected for number of features/configurations for xversion to project?

Reply:
Yes, makes sense.  If the overhead is minimal, it probably doesn't matter one way or the other.

Reply:
Ok, so it seems that it is good as it is now. Closing this review part.